src/Core/src/HotReload/AppHostBuilderExtensions.cs

```diff
@@ -0,0 +1,66 @@
+#nullable enable
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Maui.HotReload;
+
+namespace Microsoft.Maui.Hosting
+{
+	public static partial class AppHostBuilderExtensions
+	{
+		public static IAppHostBuilder EnableHotReload(this IAppHostBuilder builder, string? ideIp = null, int idePort = 9988)
+		{
+			builder.ConfigureServices<HotReloadBuilder>(hotReload =>
+			{
+				hotReload.IdeIp = ideIp;
+				hotReload.IdePort = idePort;
+			});
+			return builder;
+		}
+
+		class HotReloadBuilder : IMauiServiceBuilder
+		{
+			public string? IdeIp { get; set; }
+
+			public int IdePort { get; set; } = 9988;
+
+			public void ConfigureServices(HostBuilderContext context, IServiceCollection services)
+			{
+			}
+
+			public async void Configure(HostBuilderContext context, IServiceProvider services)
```

Comment:
This code now only runs when the builder is set up and all the bits are ready. This is before the UI is created, but the services and configuration is all set.